3 - Assignment/240521 - Rust Basic + Roguelike Game, Week 4/circular-buffer/src/lib.rs

```diff
@@ -18,29 +15,52 @@ pub enum Error {
 
 impl<T> CircularBuffer<T> {
     pub fn new(capacity: usize) -> Self {
-        unimplemented!(
-            "Construct a new CircularBuffer with the capacity to hold {}.",
-            match capacity {
-                1 => "1 element".to_string(),
-                _ => format!("{} elements", capacity),
-            }
-        );
+        let mut buf = Vec::with_capacity(capacity);
+        for _ in 0..capacity {
+            buf.push(None);
+        }
+        Self {
+            buffer: buf,
+            start_idx: 0,
+            end_idx: 0,
+        }
```

Comment:
버퍼를 한 줄에 만드는 방법도 있습니다.
```suggestion
        Self {
            buffer: (0..capacity).map(|_| None).collect(),
            start_idx: 0,
            end_idx: 0,
        }
```